third_party/move/move-binary-format/src/file_format.rs

```diff
@@ -1340,501 +1341,1256 @@ pub struct CodeUnit {
 ///
 /// Bytecodes operate on a stack machine and each bytecode has side effect on the stack and the
 /// instruction stream.
+#[bytecode_spec]
 #[derive(Clone, Hash, Eq, VariantCount, PartialEq)]
 #[cfg_attr(any(test, feature = "fuzzing"), derive(proptest_derive::Arbitrary))]
 #[cfg_attr(any(test, feature = "fuzzing"), proptest(no_params))]
 #[cfg_attr(feature = "fuzzing", derive(arbitrary::Arbitrary))]
 pub enum Bytecode {
-    /// Pop and discard the value at the top of the stack.
-    /// The value on the stack must be an copyable type.
-    ///
-    /// Stack transition:
-    ///
-    /// ```..., value -> ...```
+    #[group = "stack_and_local"]
+    #[description = "Pop and discard the value at the top of the stack. The value on the stack must be an copyable type."]
+    #[semantics = "stack >> _"]
+    #[paranoid_post = r#"
+        ty_stack >> ty
+        assert ty has drop
+    "#]
     Pop,
-    /// Return from function, possibly with values according to the return types in the
-    /// function signature. The returned values are pushed on the stack.
-    /// The function signature of the function being executed defines the semantic of
-    /// the Ret opcode.
-    ///
-    /// Stack transition:
-    ///
-    /// ```..., arg_val(1), ..., arg_val(n) -> ..., return_val(1), ..., return_val(n)```
+
+    // TODO: double check semantics
```

Comment:
Note: it's been some time since I wrote the semantics here so I'd like to double check it. Will get to it a bit later but will definitely get it done before landing.